akd/agents/search/deep_search.py

```diff
@@ -410,147 +353,59 @@ async def _execute_searches(
         queries: List[str],
         original_query: str | None = None,
         is_reformulated: bool = False,
-    ) -> List[DeepSearchResultItem]:
+    ) -> List[SearchResultItem]:
         """Execute searches using available search tools."""
         all_results = []
 
-        search_input = self.search_tool.input_schema(
-            queries=queries,
-            max_results=20,
-            category="science",
-        )
-
-        primary_results = await self.search_tool.arun(search_input)
-        all_results.extend(primary_results.results)
+        # Use primary search tool (SearchPipeline)
+        tasks: List[asyncio.Task] = []
+        tool_names: List[str] = []
 
-        # Search with Semantic Scholar if enabled
-        if self.semantic_scholar_tool and self.config.use_semantic_scholar:
-            ss_input = SemanticScholarSearchToolInputSchema(
+        # Primary search tool
+        try:
+            tool_input = self.search_tool.input_schema(
```

Comment:
maybe we can extra params like `is_reformulated`, `reformulated_query` etc as `kwargs` for time being?

Reply:
Update: In the search pipeline, now we access these params via `kwargs`:
[see](https://github.com/NASA-IMPACT/accelerated-discovery/blob/enhance/search-pipeline/akd/tools/search/pipeline.py#L569)

Reply:
Okay i have passed this as kwargs to search_tool and search pipeline in my new branch automatically accepts that now.